src/main/scala/nl/knaw/dans/easy/pid/generator/PidFormatter.scala

```diff
@@ -56,16 +58,18 @@ object PidFormatter {
    *                be mapped to chars normally not used with the given radix. The radix
    *                should therefore be sufficiently small to have enough unused chars.
    * @param dashPos position to insert a dash for readability
-   * @param pid     the PID number to format
+   * @param seed    the seed to format
    * @return the formatted PID
    */
-  def format(prefix: String, radix: Int, len: Int, charMap: Map[Char, Char], dashPos: Int)(pid: Long): String =
-    prefix + insertDashAt(dashPos)(convertToString(pid, radix, len, charMap))
-
-  def convertToString(pid: Long, radix: Int, length: Int, illegalCharMap: Map[Char, Char] = Map()): String = {
-    def padWithZeroes(s: String) = String.format(s"%${ length }s", s).replace(' ', '0')
+  def format(prefix: String, radix: Int, len: Int, charMap: Map[Char, Char], dashPos: Int)(seed: Seed): String =
+    prefix + insertDashAt(dashPos)(convertToString(seed, radix, len, charMap))
 
-    padWithZeroes(java.lang.Long.toString(pid, radix).toLowerCase).map(c => illegalCharMap.getOrElse(c, c))
+  def convertToString(seed: Seed, radix: Int, length: Int, illegalCharMap: Map[Char, Char] = Map()): String = {
+    // formats the seed in base `radix`, pads it with 0's and removes any illegal characters
```

Comment:
Why do you have to remove illegal characters? Wouldn't that be some input error or illegal state?

Reply:
nope, this has always been part of the algorithm of formatting a seed into a pid.
Basically you write the seed in a certain base (binary, hex, base64, whatever), pad it with `0` if necessary and then replace characters you don't want, as defined in `ApplicationWiring`. Why these specific characters were chosen is beyond me. What I see is that this replacements is only done for the DOI, so maybe it has something to do with its specs? Maybe @janvanmansum or @lindareijnhoudt knows more about this?